examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -124,16 +124,24 @@ void BdxOtaSender::HandleTransferSessionOutput(TransferSession::OutputEvent & ev
 
         break;
     }
-    case TransferSession::OutputEventType::kQueryReceived: {
+    case TransferSession::OutputEventType::kQueryReceived:
+    case TransferSession::OutputEventType::kQueryWithSkipReceived: {
         TransferSession::BlockData blockData;
         uint16_t blockSize   = mTransfer.GetTransferBlockSize();
         uint16_t bytesToRead = blockSize;
+        uint32_t bytesToSkip = 0;
+
+        if (event.EventType == TransferSession::OutputEventType::kQueryWithSkipReceived)
+        {
+            bytesToSkip = static_cast<uint32_t>(event.bytesToSkip.BytesToSkip);
```

Comment:
BytesToSkip is uint64_t, right? If so should we at least assert that it would fix into uint32_t?

Reply:
Indeed, BytesToSkip is uint64_t, but mNumBytesSent is uint32_t. I think it makes sense to change it to uint64_t as well - to make it as future proof as possible :) What do you think ?

Reply:
On a second thought, if we take into account that 4 bytes for the image size mean 4.194.303 blocks of maximum size 1024 bytes, and that it takes at least (most optimistically) 100msec to transfer one block - realistically one sec or more, we end up with transfer time of minimum 419430 seconds for an image with size fitting 4 bytes. This is 116 hours - which I don't think it a delay somebody is willing to wait for a new software version update :) So just an assert that it fits 4 bytes and doesn't overflow should be enough.

Reply:
SGTM. Sorry for the delayed response, my email filters seems to have stopped working :(

Reply:
I have set it to uint64_t, to simplify the logic.